addons/loyalty/models/base_partner_merge.py

```diff
@@ -0,0 +1,45 @@
+from odoo import models
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = 'base.partner.merge.automatic.wizard'
+
+    def _update_foreign_keys(self, src_partners, dst_partner):
+        """ Override of base to merge corresponding nominative loyalty cards."""
+        self._merge_loyalty_cards(src_partners, dst_partner)
+        super()._update_foreign_keys(src_partners, dst_partner)
+
+    def _merge_loyalty_cards(self, src_partners, dst_partner):
+        """ Merge nominative loyalty cards.
+
+        :param src_partners: recordset of source res.partner records to merge
+        :param dst_partner: destination res.partner record
+        """
+        LoyaltyCard = self.env['loyalty.card']
```

Comment:
```suggestion
        LoyaltyCard = self.env['loyalty.card'].sudo()
```
The wizard is available to `group_partner_manager`, which is given by default to all users, but I doubt all users have access to loyalty models by default.